### DISCUSSION TITLE
Inline filtering?

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uBlock-issues#1915] [Q&A]
Or when you decide to go into spectator after being far into the end game only to find your mates have done some ‘random direction unlit tunnels near bedrock’ smh

**Comment by u-RraaLL** [comment]
I don't think you've explained what the problem is...

But here are your mistakes:

> `www.example.com##+js(ra, aspNetHidden, div.class)`

`www.example.com##+js(ra, class, .aspNetHidden)`

> `www.example.com##.aspNetHidden.subject:remove()`

`www.example.com##.aspNetHidden:remove()`

> `www.example.com##body > div:last-of-type span:has-text(aspNetHidden)`

`aspNetHidden` is not text. `:has-text()` targets plain text in an element. In your case the node you pasted only has a space inside.

> `www.example.com##+js(rc, class, aspNetHidden, stay)`

`www.example.com##+js(rc, aspNetHidden, , stay)`

**Reply by u-RraaLL** [reply]
uBO can't cause such redirects. You might be better off trying Redirector.

Anyway, it's gonna be difficult to say more without seeing the website.

**Comment by ghost** [comment]
You need to find out first what is triggering the redirection. It can be caused by server 30x redirect, so check devtools network panel for document response header status and location. You can also check here for Refresh header. Then check page html source header for meta elements with http-equiv=refresh attribute. Then search the source for javascript code with this path being assigned to something and eventually window.location changes. If it's in the form code you posted, check for javacript which is responsible for auto-submitting it or modifying the action attribute. If you find it, then we can tell you how this can be fixed (if it's possible).

***

And there is also history(?) api which can change the address without any navigation at all.

**Reply by ghost** [reply]
> `example.com/main.aspx:50 (document)` is sending to `Location: example.com/blah/blah.aspx` when loggin in.

So it's server side redirection. You can block it but you will most likely end with empty page.

**Reply by ghost** [reply]
uBlock-user answer.

**Comment by uBlock-user** [comment]
You can try `example.com##^responseheader(location)`

**Comment by u-RraaLL** [comment]
Another attempt of the discussion's removal...
